### ISSUE TITLE
Image upload error after latest malware update

### ISSUE THREAD

**Issue by audpxi** [ultimatemember/ultimatemember#501]
### Ultimate Member Version
Tell us what UM core version you use. 
- [ ] 1.3.x
- [x] 2.x ( latest )

### Subject of the issue
After updating to latest version (which fixed this from our site: https://wordpress.org/support/topic/malicious-files-in-ultimate-members-plugin/) we can't upload images to UM profiles. Upload gets stuck in image upload window, javascript error:

```
SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at Function.n.parseJSON (jquery.js?ver=1.12.4:4)
    at Function.a.parseJSON (jquery-migrate.min.js?ver=1.4.1:2)
    at Object.onSuccess (um-functions.min.js?ver=4.9.8:1)
    at Object.success (um-fileupload.min.js?ver=4.9.8:1)
    at Object.t.success (um-jquery-form.min.js?ver=4.9.8:1)
    at i (jquery.js?ver=1.12.4:2)
    at Object.fireWith [as resolveWith] (jquery.js?ver=1.12.4:2)
    at y (jquery.js?ver=1.12.4:4)
    at XMLHttpRequest.c (jquery.js?ver=1.12.4:4) "<!DOCTYPE html>
<html lang="fi" xmlns:og="http://opengraphprotocol.org/schema/" xmlns:fb="http://www.facebook.com/2008/fbml" prefix="og: http://ogp.me/ns#">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
.... page source code continues
```

This error comes right after image is uploaded.

I also confirmed the same issue on our dev server, updated only "UM" to the newest version (Version 2.0.24) and tested image uploading before and after update. Upload worked before plugin update.

Web server uploads directory and ultimate member directory has permissions "755":
[Image](https://i.imgur.com/AFu6Ddq.png)

### Steps to reproduce the behavior
Tell us how to reproduce this issue. 

## PHP version
- [x] 7.1 above
- [ ] 5.4 
- [ ] 5.3 below

### Do you use UM extensions? 
- [ ] Private Content 
- [ ] Instagram
- [ ] User Tags 
- [ ] Social Activity
- [ ] WooCommerce
- [ ] Private Messages
- [ ] Followers
- [ ] Real-time Notifications
- [ ] Social Login
- [ ] bbPress
- [ ] MailChimp
- [ ] User Reviews
- [ ] Verified Users
- [ ] myCRED
- [ ] Notices
- [ ] Profile Completeness 
- [ ] Friends
- [ ] Terms & Conditions
- [ ] Google reCAPTCHA 
- [ ] Online Users
- [ ] Groups
- [ ] Gallery

**Comment by audpxi** [comment]
Just tested version 2.0.21, which works normally. Malware is not fixed in that version, so can't use that in production...

**Comment by gitblair** [comment]
@champsupertramp I'm up to the most recent version but I'm still having an issue allowing people to upload a profile photo during registration. I was working fine, with your special snippet, prior to all this hacking business. Now what happens is the profile photo is uploaded during registration but it is moved into their UM upload folder as a "stream_photo_" with so that in order for it to display as a profile photo, now I need to manually rename the "stream_photo_" to profile_photo.jpg. Is this a known issue being worked or should I abandon the idea of allowing profile photo uploads on the registration form?

Thank you

**Comment by audpxi** [comment]
Got jQuery error (SyntaxError: Unexpected token < in JSON at position 0) fixed with updating to 2.0.25. But images are uploaded as "stream_photo_", just like @gitblair told. Profile thumbnail compression is not working anymore?

Settings: https://imgur.com/xXI5jBy
Before: https://imgur.com/vQWuQSe
Now: https://imgur.com/aHlhWfU

**Comment by jakegiltsoff** [comment]
Did anyone find a fix for this?
